src/components/layout.tsx

```diff
@@ -25,25 +31,37 @@ const Layout = ({ children, title, description, img }: Props) => {
     }
   });
 
-  const setupObserver = () => {
-    const root = document.querySelector('.side-nav');
-    const options = { root, threshold: 0.5, rootMargin: '-93px 0px 0px 0px' };
-    const observer = new IntersectionObserver(onIntersectionChange, options);
-    const targets = document.querySelectorAll('.side-nav__title');
-    targets.forEach((target: Element) => observer.observe(target));
-  };
+  const setupObserver = (): void => {
+    const container = document.querySelector('.side-nav') as HTMLElement;
+    const stickyElementsClassName = 'side-nav__title';
+    const root = isScreenWithinWidth(MAX_MOBILE_SCREEN_WIDTH)
+      ? null
+      : container;
+    const headerRootMargin = '-93px 0px 0px 0px';
```

Comment:
Trivial nit: use type for variable?

```suggestion
    const headerRootMargin: string = '-93px 0px 0px 0px';
```

Reply:
Added types for this and everything else around it. :tada: